base/applications/rapps/appview.cpp

```diff
@@ -1697,9 +1697,44 @@ CApplicationView::SetRedraw(BOOL bRedraw)
 }
 
 void
-CApplicationView::SetFocusOnSearchBar()
+CApplicationView::SetFocusOnSearchBar(UINT SetSel)
 {
     m_SearchBar->SetFocus();
+    if (SetSel)
+    {
+        m_SearchBar->SendMessage(EM_SETSEL, (INT8)LOBYTE(LOWORD(SetSel)), (INT8)HIBYTE(LOWORD(SetSel)));
+        m_SearchBar->SendMessage(EM_SETSEL, (INT8)LOBYTE(HIWORD(SetSel)), (INT8)HIBYTE(HIWORD(SetSel)));
+    }
+}
+
+void
+CApplicationView::SetSearchText(LPCWSTR Str)
+{
+    m_SearchBar->SetWindowText(Str);
+}
+
+void
+CApplicationView::SelectItem(CAppInfo *pAI)
+{
+    int i = -1;
```

Comment:
```suggestion
    int i;
```
This extra initialization is not needed since in all cases (the if and the else blocks) `i` gets initialized to something.